kolibri2zim/scraper.py

```diff
@@ -880,18 +932,21 @@ def sanitize_inputs(self):
         channel_meta = self.db.get_channel_metadata(self.channel_id)
 
         # input  & metadata sanitation
-        period = datetime.datetime.now().strftime("%Y-%m")
+        period = datetime.datetime.now(datetime.UTC).strftime("%Y-%m")
         if self.fname:
             # make sure we were given a filename and not a path
-            self.fname = Path(self.fname.format(period=period))
-            if Path(self.fname.name) != self.fname:
-                raise ValueError(f"filename is not a filename: {self.fname}")
+            fname_path = Path(self.fname.format(period=period)).resolve()
+            if not fname_path.is_file():
+                msg = f"filename is not a filename: {self.fname}"
+                raise ValueError(msg)
+            self.clean_fname = fname_path.as_posix()
         else:
-            self.fname = f"{self.name}_{period}.zim"
+            self.clean_fname = f"{self.name}_{period}.zim"
 
         if not self.title:
-            self.title = channel_meta["name"]
-        self.title = self.title.strip()
+            self.clean_title = channel_meta["name"].strip()
```

Comment:
- Why would we need a `clean_title` **and** a `title`?
- btw this `self.clean_title` is not used so `self.title` remains missing.
- How is this better than the previous code?

Reply:
Same as before, type checker consider that self.title might be None, while `self.clean_title` is a string for the type checker.
The fact that `self.title` might be None was an issue later on for the type checker, but it is not anymore, so I reverted my change.
But you are right that I should have use `self.clean_title` afterwards 😯 

PS: I did not clicked "Resolve conversation", is it better in the "code" view?

Reply:
PS : It looks like it is not better, you just see that it is outdated but you do not see my new code ...